native/result/bindings.go

```diff
@@ -0,0 +1,77 @@
+package result
+
+import (
+	"fmt"
+	vm "github.com/goby-lang/goby/vm"
+	errors "github.com/goby-lang/goby/vm/errors"
+)
+
+func init() {
+	vm.RegisterExternalClass(
+		"result", vm.ExternalClass(
+			"Result",
+			"result.gb",
+			map[string]vm.Method{
+				"empty": _binding_Result_Empty,
+				"new":   _binding_Result_New,
+			},
+			map[string]vm.Method{
+				"method_missing": _binding_Result_MethodMissing,
+				"or":             _binding_Result_Or,
+			}))
+}
+
+var _staticResult = new(Result)
+
+func _binding_Result_New(receiver vm.Object, line int, t *vm.Thread, args []vm.Object) vm.Object {
+	r := _staticResult
+	if len(args) != 2 {
+		return t.VM().InitErrorObject(errors.ArgumentError, line, errors.WrongNumberOfArgumentFormat, 2, len(args))
+	}
+	arg0, ok := args[0].(Object)
+	if !ok {
+		panic("Argument 0 must be Object")
+	}
+
+	arg1, ok := args[1].(Object)
+	if !ok {
+		panic("Argument 1 must be Object")
+	}
+
+	return r.New(t, arg0, arg1)
+}
+
+func _binding_Result_Empty(receiver vm.Object, line int, t *vm.Thread, args []vm.Object) vm.Object {
+	r := _staticResult
+	if len(args) != 0 {
+		return t.VM().InitErrorObject(errors.ArgumentError, line, errors.WrongNumberOfArgumentFormat, 0, len(args))
+	}
+	return r.Empty(t)
+}
+
+func _binding_Result_MethodMissing(receiver vm.Object, line int, t *vm.Thread, args []vm.Object) vm.Object {
+	r, ok := receiver.(*Result)
+	if !ok {
+		panic(fmt.Sprintf("Impossible receiver type. Wanted Result got %s", receiver))
+	}
+	if len(args) != 1 {
+		return t.VM().InitErrorObject(errors.ArgumentError, line, errors.WrongNumberOfArgumentFormat, 1, len(args))
+	}
+	arg0, ok := args[0].(Object)
+	if !ok {
+		panic("Argument 0 must be Object")
+	}
+
+	return r.MethodMissing(t, arg0)
+}
+
+func _binding_Result_Or(receiver vm.Object, line int, t *vm.Thread, args []vm.Object) vm.Object {
```

Comment:
@eliothedeman will we change the function name to use more Go-like naming?

Reply:
This is tough, because I don't want to have to build logic for handling name conflicts. So using a name like this makes it pretty likely that won't happy.